File: bms/go/go-gt/NewMatrix/NewMatrix.go
package test

var ()

const ()

func NewMatrix(n int64) (m *Matrix) {
	m = new(Matrix)
	m.N = n
	m.A = make([]int64, n*n)
	return m
}

type Matrix struct {
	N int64
	A []int64
}

File: translations_only/fail/go/go-gt/NewMatrix/claude2/Hinted/NewMatrix.rs

use std::boxed::Box;

struct Matrix {
    n: usize,
    a: Box<[i64]>,
}

fn new_matrix(n: usize) -> Box<Matrix> {
    let a = vec![0; n * n].into_boxed_slice();

    Box::new(Matrix {
        n,
        a,
    })
}
